### PR TITLE
Enable evaluation of all ternary ops in `ExpressionEvaluator`

### PR DESCRIPTION
**Pull request by jacobhinkle**
This makes `clamp`, `lerp`, `threshold`, and `where` evaluable by the `ExpressionEvaluator` with or without `PrecomputedValues`.

### CODE REVIEW

**Comment by jacobhinkle** [comment]
!build

File: csrc/evaluator_common.cpp [hunk 1]
```diff
@@ -435,14 +443,49 @@ void NaiveValueMachine::makeBinaryOp(BinaryOp* bop) {
...
+
+  NVF_ERROR(in >= 0, "Integer Machine: unknown input: ", lsop);
+  NVF_ERROR(out >= 0, "Integer Machine: unknown out: ", lsop);
+
+  int index = makeInstructionEntry();
+  inst_type_[index] = InstructionType::SET_OP;
+  src0_[index] = in;
+  dest_[index] = out;
+}
+
```
Comment (jacobhinkle) [inline csrc/evaluator_common.cpp:478]: I can make this a separate PR also, but it's very small since half of the machinery for `set` was already in place.

**Comment by jacobhinkle** [comment]
Test failures are unrelated: random div op test violated tolerance and codegen diff index non-determinism.

**Review by zasdfgbnm — APPROVED** [review]
Could you check CPU overhead? I don't think it will make different, but just to confirm.

**Comment by jacobhinkle** [comment]
> Could you check CPU overhead? I don't think it will make different, but just to confirm.

Is there a benchmark you've used to check this for `ExpressionEvaluator` before?

**Comment by zasdfgbnm** [comment]
> > Could you check CPU overhead? I don't think it will make different, but just to confirm.
> 
> Is there a benchmark you've used to check this for `ExpressionEvaluator` before?

I am talking about the same benchmark as https://github.com/NVIDIA/Fuser/pull/876#issuecomment-1722523249

**Comment by jacobhinkle** [comment]
> > > Could you check CPU overhead? I don't think it will make different, but just to confirm.
> > 
> > 
> > Is there a benchmark you've used to check this for `ExpressionEvaluator` before?
> 
> I am talking about the same benchmark as [#876 (comment)](https://github.com/NVIDIA/Fuser/pull/876#issuecomment-1722523249)

Looks like no change:
```
[----------------------------------------  ---------------------------------------]
           |  " 2023-09-19 5ff26b9e"  |  "evaluate_ternary_ops 2023-09-19 b2897a6e"
1 threads: ------------------------------------------------------------------------
      2    |           17.4           |                     17.5                   
      4    |           21.5           |                     22.3                   
      8    |           31.5           |                     31.7                   
      16   |           50.6           |                     51.3                   
      32   |           95.1           |                     95.6                   
      64   |          184.8           |                    186.2                   
      128  |          407.8           |                    401.6                   

Times are in microseconds (us)
```

**Pull request merged**